backend/certificates.js

```diff
@@ -0,0 +1,173 @@
+import {getSetting, saveSetting} from './settings.js';
+import * as fs from 'fs';
+import fetch from 'node-fetch';
+import Logger from './logger';
+
+import {fakeCerts} from '../test/backend/utils';
+
+const CA_HOST = {
+    PROTOCOL: 'http',
+    DOMAIN: 'plotly-connector-test.com',
+    ROUTE: 'certificate'
+};
+
+const daysCertificateIsGoodFor = 24;
+
+export const LOCAL_SETTINGS = {
+    CA_HOST_URL: `${CA_HOST.PROTOCOL}://${CA_HOST.DOMAIN}/${CA_HOST.ROUTE}`,
+    MAX_TRIES_COUNT: 5,
+    ONGOING_COUNT: 0,
+    TIMEOUT_BETWEEN_TRIES: 1,  // seconds
+    USE_MOCK_CERTS: false
+};
+
+export function setCertificatesSettings(setting, value) {
+    LOCAL_SETTINGS[setting] = value;
+}
+
+// return HTTPS certs if they exist for a server to use when created or null
+export function getCerts() {
+    try {
+        return {
+            key: fs.readFileSync(getSetting('KEY_FILE'), 'utf-8'),
+            cert: fs.readFileSync(getSetting('CERT_FILE'), 'utf-8')
+        };
+    } catch (e) {
+        return {};
+    }
+}
+
+export function saveCertsLocally({key, cert, subdomain}) {
+    // (renewal) -> fs.writeFile replaces the file if it already exists
+    const saveCert = new Promise((resolve, reject) => {
+        fs.writeFile(getSetting('CERT_FILE'), cert, (err) => {
+            if (err) reject(err);
+            else resolve();
+        });
+    });
+    const saveKey = new Promise((resolve, reject) => {
+        fs.writeFile(getSetting('KEY_FILE'), key, (err) => {
```

Comment:
I wanted to return a `promise.all` so that I can do a .then on it form the parent function(s)